### ISSUE TITLE
DevHome GitHub extension upgrade stuck in a loop

### ISSUE THREAD

**Issue by davidegiacometti** [microsoft/devhome#2198]
### Dev Home version

0.1001.389.0

### Windows build number

10.0.19045.0

### Steps to reproduce the bug

I have upgraded DevHome and GitHub extension to latest version from the Store but the upgrade is stuck in the below loop:

1. GitHub extension upgrade failed.

![image](https://github.com/microsoft/devhome/assets/25966642/3e4cbfd8-0878-4763-bb73-d39b7ad3d128)

2. Using the `wscollect` command I have been able to determine why the installation failed: it's related to a dependency of the MSIX.

```
AppX Deployment operation failed for package Microsoft.WindowsAppRuntime.1.4_4000.1082.2259.0_x64__8wekyb3d8bbwe with error 0x80073D02. The specific error text for this failure is: error 0x80073D02: Unable to install because the following apps need to be closed Microsoft.Windows.DevHome_0.1001.389.0_x64__8wekyb3d8bbwe Microsoft.YourPhone_1.23112.87.0_x64__8wekyb3d8bbwe.
``` 

3. After killing `PhoneExperienceHost.exe` process I have been able to upgrade the GitHub extension from the Store.

4. Start DevHome, the "Dev Home is getting things ready for you..." window appear and the GitHub extension is downloaded again stucking at 95%.

5. Finally DevHome starts but widget are not working.

![image](https://github.com/microsoft/devhome/assets/25966642/ce564f3a-caa9-4fdf-beb6-dba67ad3531c)

6. Very strange that the package resulted as installed.

```
PS C:\Users\david> Get-AppxPackage -Name Microsoft.Windows.DevHomeGitHubExtension

Name              : Microsoft.Windows.DevHomeGitHubExtension
Publisher         : CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US
Architecture      : X64
ResourceId        :
Version           : 0.1000.393.0
PackageFullName   : Microsoft.Windows.DevHomeGitHubExtension_0.1000.393.0_x64__8wekyb3d8bbwe
InstallLocation   : C:\Program Files\WindowsApps\Microsoft.Windows.DevHomeGitHubExtension_0.1000.393.0_x64__8wekyb3d8bbwe
IsFramework       : False
PackageFamilyName : Microsoft.Windows.DevHomeGitHubExtension_8wekyb3d8bbwe
PublisherId       : 8wekyb3d8bbwe
IsResourcePackage : False
IsBundle          : False
IsDevelopmentMode : False
NonRemovable      : False
Dependencies      : {Microsoft.WindowsAppRuntime.1.4_4000.1082.2259.0_x64__8wekyb3d8bbwe}
IsPartiallyStaged : False
SignatureKind     : Store
Status            : Ok
```

### Expected result

DevHome and its extensions have been updated successfully.

### Actual result

Upgrade is stuck in a loop.

**Comment by krschau** [comment]
I think logs would be useful for this one. Can you also tell me the result if you run `get-appxpackage *widget*` in Powershell?

/logs

**Comment by davidegiacometti** [comment]
Update: I have removed the `Microsoft.WidgetsPlatformRuntime` package and reinstalled it from DevHome.
Now DevHome is starting as expected and widget are working.

Note that the packages below are still present in the Store as not updated.
The issue may not related to DevHome itself: feel free to close it.

![image](https://github.com/microsoft/devhome/assets/25966642/a49a93e6-8819-4083-91cd-5ad6ae3da8fe)

**Comment by krschau** [comment]
I've been keeping an eye on this and I'm inclined to agree with you that it was store flakiness. Especially, as you point out in your last comment, since the Store claimed there was still an update available while 1.1.0.0 (the version you have) is the most recent version. The next release of Dev Home will include #2226, which should at least prevent the weird loop in the case of more store problems. Please let us know if you see any more weirdness here.
